phpBB/phpbb/storage/controller/attachment.php

```diff
@@ -0,0 +1,591 @@
+				$post_row = array('forum_id' => false);
+				$this->phpbb_download_handle_pm_auth( $attachment['post_msg_id']);
+			}
+
+			$extensions = array();
+			if (!extension_allowed($post_row['forum_id'], $attachment['extension'], $extensions))
+			{
+				throw new http_exception(403, 'EXTENSION_DISABLED_AFTER_POSTING', [$attachment['extension']]);
+			}
+		}
+
+		$display_cat = $extensions[$attachment['extension']]['display_cat'];
+
+		if (($display_cat == ATTACHMENT_CATEGORY_IMAGE || $display_cat == ATTACHMENT_CATEGORY_THUMB) && !$this->user->optionget('viewimg'))
+		{
+			$display_cat = ATTACHMENT_CATEGORY_NONE;
+		}
+
+		if ($display_cat == ATTACHMENT_CATEGORY_FLASH && !$this->user->optionget('viewflash'))
+		{
+			$display_cat = ATTACHMENT_CATEGORY_NONE;
+		}
+
+		if ($thumbnail)
+		{
+			$attachment['physical_filename'] = 'thumb_' . $attachment['physical_filename'];
+		}
+		else if ($display_cat == ATTACHMENT_CATEGORY_NONE && !$attachment['is_orphan'])
+		{
+			// Update download count
+			$this->phpbb_increment_downloads($attachment['attach_id']);
+		}
+
+		$redirect = '';
+
+		/**
+		* Event to modify data before sending file to browser
+		*
+		* @event core.download_file_send_to_browser_before
+		* @var	int		attach_id			The attachment ID
+		* @var	array	attachment			Array with attachment data
+		* @var	int		display_cat			Attachment category
+		* @var	array	extensions			Array with file extensions data
+		* @var	string	mode				Download mode
+		* @var	bool	thumbnail			Flag indicating if the file is a thumbnail
+		* @var	string	redirect			Do a redirection instead of reading the file
+		* @since 3.1.6-RC1
+		* @changed 3.1.7-RC1	Fixing wrong name of a variable (replacing "extension" by "extensions")
+		* @changed 3.3.0-a1		Add redirect variable
+		*/
+		$vars = array(
+			'attach_id',
+			'attachment',
+			'display_cat',
+			'extensions',
+			'mode',
+			'thumbnail',
+			'redirect',
+		);
+		extract($this->dispatcher->trigger_event('core.download_file_send_to_browser_before', compact($vars)));
+
+		if (!empty($redirect))
+		{
+			$response = new RedirectResponse($redirect);
+			$response->send();
+		}
+		else
+		{
+			$this->send_file_to_browser($attachment, $display_cat);
+		}
+	}
+
+	/**
+	* Send file to browser
+	*/
+	protected function send_file_to_browser($attachment, $category)
+	{
+		$filename = $attachment['physical_filename'];
+
+		if (!$this->storage->exists($filename))
+		{
+			throw new http_exception(404, 'ERROR_NO_ATTACHMENT');
+		}
+
+		// Correct the mime type - we force application/octetstream for all files, except images
+		// Please do not change this, it is a security precaution
+		if ($category != ATTACHMENT_CATEGORY_IMAGE || strpos($attachment['mimetype'], 'image') !== 0)
+		{
+			$attachment['mimetype'] = (strpos(strtolower($this->user->browser), 'msie') !== false || strpos(strtolower($this->user->browser), 'opera') !== false) ? 'application/octetstream' : 'application/octet-stream';
+		}
+
+		if (@ob_get_length())
```

Comment:
@marc1706

Reply:
@Nicofuma

Reply:
well I saw it when I reviewed your PR and asked myself if it was useful. The fact is that I don't know.... (and we can't really do what I do at work in this case: removes it and wait to see if someone cries =D)

Reply:
:joy:

Reply:
https://github.com/phpbb/phpbb/commit/be3bdf61d778928916942c07c6d60968fddf9d18
https://github.com/phpbb/phpbb/commit/20d18e1a9fec071d6e2e519709b08bacf650d750

Reply:
so useful commits....